src/osdc/Objecter.cc

```diff
@@ -1319,18 +1320,20 @@ void Objecter::handle_osd_map(MOSDMap *m)
     }
     sl.unlock();
     put_session(s);
+    s = nullptr;
   }
   for (list<LingerOp*>::iterator p = need_resend_linger.begin();
        p != need_resend_linger.end(); ++p) {
     LingerOp *op = *p;
     if (!op->session) {
       _calc_target(&op->target, nullptr);
-      OSDSession *s = NULL;
+      OSDSession *s = nullptr;
       const int r = _get_session(op->target.osd, &s, sul);
       assert(r == 0);
-      assert(s != NULL);
+      assert(s != nullptr);
```

Comment:
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions